llvm/lib/Target/AIE/AIE2InstructionSelector.cpp

```diff
@@ -4140,6 +4266,168 @@ bool AIE2InstructionSelector::select512BitG_AIE_STORE_SRS(
   }
 }
 
+// Returns the instruction where the Reg is defined ignoring G_BITCAST.
+static MachineInstr *stripBitCast(Register Reg, MachineRegisterInfo &MRI) {
+  MachineInstr *MI = getDefIgnoringCopies(Reg, MRI);
+  if (MI->getOpcode() == TargetOpcode::G_BITCAST)
+    return getDefIgnoringCopies(MI->getOperand(1).getReg(), MRI);
+  return MI;
+}
+
+// If two 256-bit SRS are input to 512-bit STORE through CONCAT, these are
+// combined into VST.SRS
+bool AIE2InstructionSelector::selectG_AIE_STORE_CONCAT_SRS(
+    MachineInstr &StoreI, MachineRegisterInfo &MRI) {
+
+  Register ConcatResult = (StoreI.uses().begin())->getReg();
+  MachineInstr *ConcatOp = MRI.getUniqueVRegDef(ConcatResult);
+
+  assert(ConcatOp && "Expected SSA.");
+
+  if (!canDelayMemOp(*ConcatOp, StoreI, MRI))
+    return false;
```

Comment:
Can you add a comment to explain what this is checking? I think we are verifying that it is safe to delay the VSRS operation. In particular, we verify that the control registers aren't clobbered in between the vsrs and the store.